File: python/pig_latin_test.py
"""
Tests for the pig latin translator
"""
from unittest import TestCase, main

from pig_latin import translate

class TranslateTestCases(TestCase):
    """
    Tests the translate function for pig latin
    """
    def test_translates_word_beginning_with_a_vowel(self):
        """
        adds ay
        """
        res = translate('apple')
        expected = "appleay"
        self.assertEqual(res, expected)

    def test_retains_punctuation(self):
        """
        does not remove punctuation marks
        """
        res = translate('apples!')
        expected = "applesay!"
        self.assertEqual(res, expected)

    def test_retains_capitalization(self):
        """
        if given a capital lettered word, it returns it the same
        """
        res = translate('Apples')
        expected = "Applesay"
        self.assertEqual(res, expected)

    def test_translates_consonant_words(self):
        """
        can check for single consnants
        """
        res = translate('banana')
        expected = "ananabay"
        self.assertEqual(res, expected)

    def test_translates_words_with_multiple_consonants(self):
        """
        like school, cherry
        """
        res = translate('school')
        expected = 'oolschay'
        self.assertEqual(res, expected)

    def test_translates_qu_words(self):
        """
        qu should get treated as a single phoneme
        """
        res = translate('quick')
        expected = 'ickquay'
        self.assertEqual(res, expected)

    def test_translates_sentences(self):
        """
        translates all the words in a sentence
        """
        res = translate('the quick brown fox')
        expected = 'ethay ickquay ownbray oxfay'
        self.assertEqual(res, expected)

if __name__ == "__main__":
    main()

File: js/pigLatin.test.js
// Rewrite this in Unit Test

var {translate: pig} = require("./pigLatin");

// write a test asserting that capitalized words are still capitalized
// (but with a different initial capital letter, of course) retain the
// punctuation from the original phrase

test("translate a word beginning with a vowel", () => {
  expect(pig("kevin's")).toBe("evinkay's")
});

test("translate a word beginning with a vowel", () => {
  expect(pig("apple")).toBe("appleay")
});

test("translates a word beginning with a consonant:", () => {
  expect(pig("banana")).toBe("ananabay")
});

test("translates a word beginning with two consonants", () => {
  expect(pig("cherry")).toBe("errychay")
});

test("translates two words", () => {
  expect(pig("eat pie")).toBe("eatay iepay")
});

test("translates a word beginning with three consonants", () => {
  expect(pig("three")).toBe("eethray")
});

test("counts 'sch' as a single phoneme", () => {
  expect(pig("school")).toBe("oolschay")
});

test("counts 'qu' as a single phoneme", () => {
  expect(pig("quiet")).toBe("ietquay")
});

test("counts 'qu' as a consonant even when it's preceded by a consonant", () => {
  expect(pig("square")).toBe("aresquay")
});

test("translates many words", () => {
  expect(pig("the quick brown fox")).toBe("ethay ickquay ownbray oxfay")
});

test("translates words and retains punctuation", () => {
  expect(pig("the quick, brown, fox")).toBe("ethay ickquay, ownbray, oxfay")
});

test("translates words and retains punctuation and capitalization", () => {
  expect(pig("The quick, brown, fox.")).toBe("Ethay ickquay, ownbray, oxfay.")
});